src/isl/util.rs

```diff
@@ -0,0 +1,286 @@
+        RangeBoundaryValue::Value(RangeBoundaryValueType::Float(value), range_boundary_type)
+    }
+    pub fn timestamp_value(value: Timestamp, range_boundary_type: RangeBoundaryType) -> Self {
+        RangeBoundaryValue::Value(
+            RangeBoundaryValueType::Timestamp(value),
+            range_boundary_type,
+        )
+    }
+    pub fn decimal_value(value: Decimal, range_boundary_type: RangeBoundaryType) -> Self {
+        RangeBoundaryValue::Value(RangeBoundaryValueType::Decimal(value), range_boundary_type)
+    }
+
+    fn from_ion_element(value: &OwnedElement, is_non_negative: bool) -> IonSchemaResult<Self> {
+        let value_annotations: Vec<&OwnedSymbolToken> = value.annotations().collect();
+        let range_boundary_type = if value_annotations.contains(&&text_token("exclusive")) {
+            RangeBoundaryType::Exclusive
+        } else {
+            RangeBoundaryType::Inclusive
+        };
+
+        match value.ion_type() {
+            IonType::Symbol => {
+                let sym = try_to!(try_to!(value.as_sym()).text());
+                match sym {
+                    "min" => Ok(RangeBoundaryValue::Min),
+                    "max" => Ok(RangeBoundaryValue::Max),
+                    _ => {
+                        return invalid_schema_error(format!(
+                            "Range boundary value: {} is not supported",
+                            sym
+                        ))
+                    }
+                }
+            }
+            IonType::Integer => {
+                if is_non_negative {
+                    let value = Range::validate_non_negative_integer_range_boundary_value(
+                        value.as_any_int().unwrap(),
+                    )?
+                    .to_owned();
+                    Ok(RangeBoundaryValue::int_non_negative_value(
+                        value,
+                        range_boundary_type,
+                    ))
+                } else {
+                    Ok(RangeBoundaryValue::int_value(
+                        value.as_any_int().unwrap().as_i64().unwrap(),
+                        range_boundary_type,
+                    ))
+                }
+            }
+            IonType::Decimal => Ok(RangeBoundaryValue::decimal_value(
+                value.as_decimal().unwrap().to_owned(),
+                range_boundary_type,
+            )),
+            IonType::Float => Ok(RangeBoundaryValue::float_value(
+                value.as_f64().unwrap(),
+                range_boundary_type,
+            )),
+            IonType::Timestamp => Ok(RangeBoundaryValue::timestamp_value(
+                value.as_timestamp().unwrap().to_owned(),
+                range_boundary_type,
+            )),
+            _ => return invalid_schema_error("Unsupported range type specified"),
+        }
+    }
+}
+
+/// Represents the range boundary types in terms of exclusivity (i.e. inclusive or exclusive)
+#[derive(Debug, Clone, PartialEq, PartialOrd)]
```

Comment:
I had added this because of the comparison to `start > end` validation but as it depends on timestamp ordering which is not yet implemented, I can remove it for now.

Reply:
If it's something that we for sure need later, don't remove it now just because I don't understand it.

The `start > end` validation relies on comparing `RangeBoundaryValue`s or possibly `RangeBoundaryValueType`s. I understand why the `RangeBoundaryValueType` should have `PartialOrd`, but I don't understand why `RangeBoundaryType` has any ordering at all. Is `Inclusive` greater than `Exclusive` or is it the other way around?